mapeamento/mapeadores/spiders/bases/mapeador_semantico.py

```diff
@@ -0,0 +1,106 @@
+import scrapy
+
+from unidecode import unidecode
+
+from mapeadores.spiders.bases.mapeador import Mapeador
+
+class MapeadorSemantico(Mapeador):
+    custom_settings = {
+        "RETRY_ENABLED": False,
+        "AUTOTHROTTLE_ENABLED": True,
+        "AUTOTHROTTLE_TARGET_CONCURRENCY": 100,
+    }
+
+    def start_requests(self):
+        self._read_data()        
+
+        for i in range(len(self.territories)):
+            self._print_log(i)
+
+            name = self._normalize_str(self.territories[i]["name"])
+            state_code = self._normalize_str(self.territories[i]["state_code"])
+
+            item = {
+                "territory_id": self.territories[i]['id'],
+                "city": self.territories[i]['name'],
+                "state": state_code,
+            }
+
+            for protocol_option in ["http", "https"]:
+                for name_option in self.domain_generator(name):                    
+                    for url_option in self.urls_pattern(protocol_option, name_option, state_code):
+                        yield scrapy.Request(
+                            url_option, 
+                            callback=self.parse, 
+                            cb_kwargs={"item": item}
+                        )
+
+    def _read_data(self):
+        print(f"------- COMEÇANDO MAPEAMENTO DE {self.pattern_name()} -------")
+        self.read_csv() 
```

Comment:
Acho que o corpo do `read_csv` faria mais sentido sendo o método `__init__` do `Mapeador`. Assim, não entendi muito bem o propósito desse log de `COMEÇANDO MAPEAMENTO DE` já que o log padrão de `Spider opened` já demarca o início da raspagem.

Reply:
Brilhante, esqueci que tinha a opção de usar o construtor. Adotado. 
O `COMEÇANDO MAPEAMENTO` era pro caso de se executar vários mapeadores em sequência. Intermediei a situação adicionando o mapeador sendo executado na mensagem de progresso: 

![image](https://github.com/user-attachments/assets/e67f981b-d7e8-416a-803b-87e0ea6e78e2)